ftw/solr/indexer.py

```diff
@@ -18,6 +18,16 @@ class SolrIndexQueueProcessor(object):
 
     def index(self, obj, attributes=None):
         """Index the given object."""
+        if attributes:
+            attributes = tuple(attributes)
+
+        if attributes and 'path' in attributes and 'path_depth' not in attributes:
```

Comment:
I would prefer doing this in the handler's `get_data()` method.
It already does similar stuff like adding the `UID` if it's not part of `attributes` and it also ensure that only attributes are added that are in the Solr schema.

Reply:
Good point - that's a more natural fit and eliminates some of the clunky tuple-handling I had to do before. Moved it into `get_data()` 👍